### PR TITLE
[Feature][Connector-V2]  jdbc connector supports Kingbase database

### LINKED ISSUE #4512: 支持国产数据库吗，如人大金仓Kingbase
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

请问，支持国产数据库了吗？支持粒度咋样，如人大金仓Kingbase

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #4758: [Feature][Connector-V2]   Hope jdbc connector supports KingbaseES database
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

 Hope jdbc connector supports KingBase database

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by NickCodeJourney**
close #4512
close #4758

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
why close？

**Comment by NickYoungPeng** [comment]
> why close？

some code is not push, wait a min

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kingbasees/KingbaseESTypeMapper.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        switch (kbType) {
+            case KB_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case KB_BOOLEAN_ARRAY:
+                return ArrayType.BOOLEAN_ARRAY_TYPE;
+            case KB_BYTEA:
+                return PrimitiveByteArrayType.INSTANCE;
+            case KB_BYTEA_ARRAY:
+                return ArrayType.BYTE_ARRAY_TYPE;
+            case KB_SMALLINT:
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kingbasees/KingbaseESTypeMapper.java]: SMALLINT should map to short

**Comment by hailin0** [comment]
please add e2e testcase
https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

**Comment by NickYoungPeng** [comment]
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e


What  os  environment does e2e need？

**Comment by NickYoungPeng** [comment]
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

can you help me? tell me some skill how can i do it

**Comment by NickCodeJourney** [comment]
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

need all success？
Snipaste_2023-07-07_09-57-44

**Comment by NickCodeJourney** [comment]
@hailin0 @EricJoy2048 @Hisoka-X please review

**Comment by NickCodeJourney** [comment]
> > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> 
> need all success？ Snipaste_2023-07-07_09-57-44

hello @TyrantLucifer ,when i test using spark container 
`Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968"
	at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63)
	at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
	at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`

**Comment by NickCodeJourney** [comment]
> > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > 
> > 
> > need all success？ Snipaste_2023-07-07_09-57-44
> 
> hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`

image
  in local env is ok,I suspect something is wrong with the container

**Comment by CheneyYin** [comment]
> > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > 
> > > 
> > > need all success？ Snipaste_2023-07-07_09-57-44
> > 
> > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> 
> image
>   in local env is ok,I suspect something is wrong with the container

[This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)

**Comment by NickCodeJourney** [comment]
> > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > 
> > > > 
> > > > need all success？ Snipaste_2023-07-07_09-57-44
> > > 
> > > 
> > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > 
> > 
> > image
> >   in local env is ok,I suspect something is wrong with the container
> 
> [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)

I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

**Comment by CheneyYin** [comment]
> > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > 
> > > > > 
> > > > > need all success？ Snipaste_2023-07-07_09-57-44
> > > > 
> > > > 
> > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > 
> > > 
> > > image
> > >   in local env is ok,I suspect something is wrong with the container
> > 
> > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> 
> I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

I mean, you can try to configure LC=en_US.UTF-8 for spark container.  Because in the container, LC=POSIX usually.

**Comment by NickYoungPeng** [comment]
Yes, I have added the spark container, but it can't be solved

**Comment by NickCodeJourney** [comment]
> > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > 
> > > > > 
> > > > > need all success？ Snipaste_2023-07-07_09-57-44
> > > > 
> > > > 
> > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > 
> > > 
> > > image
> > >   in local env is ok,I suspect something is wrong with the container
> > 
> > 
> > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> 
> I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

Yes, I have added the spark container, but it can't be solved.

**Comment by NickCodeJourney** [comment]
> > > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > > 
> > > > > > 
> > > > > > need all success？ Snipaste_2023-07-07_09-57-44
> > > > > 
> > > > > 
> > > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > > 
> > > > 
> > > > image
> > > >   in local env is ok,I suspect something is wrong with the container
> > > 
> > > 
> > > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> > 
> > 
> > I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.
> 
> Yes, I have added the spark container, but it can't be solved.

image

**Comment by CheneyYin** [comment]
> > > > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > > > 
> > > > > > > 
> > > > > > > need all success？ Snipaste_2023-07-07_09-57-44
> > > > > > 
> > > > > > 
> > > > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > > > 
> > > > > 
> > > > > image
> > > > >   in local env is ok,I suspect something is wrong with the container
> > > > 
> > > > 
> > > > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> > > 
> > > 
> > > I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.
> > 
> > 
> > Yes, I have added the spark container, but it can't be solved.
> 
> image

```shell
 docker run tyrantlucifer/spark:2.4.6 locale -a
 05:17:56.97 
 05:17:56.97 Welcome to the Bitnami spark container
 05:17:56.98 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-spark
 05:17:56.98 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-spark/issues

 05:17:56.98 
C
C.UTF-8
POSIX
```
This container doesn't support `zh_CN.utf-8`.
And the kingbase container(`huzhihui/kingbase:v8r6`) locale `en_US.UTF-8`.

**Comment by CheneyYin** [comment]
image
It can fix `ansi_x3.4-1968` error.

**Comment by NickCodeJourney** [comment]
> image It can fix `ansi_x3.4-1968` error.

thx sir

**Comment by NickCodeJourney** [comment]
Snipaste_2023-07-13_22-25-58

**Comment by liugddx** [comment]
@zhilinli123 PTAL

**Comment by zhilinli123** [comment]
Are you willing to add connector V2 documentation for this? 
Refer to code:https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md
Issues:https://github.com/apache/seatunnel/issues/4572
Issues example:https://github.com/apache/seatunnel/issues/4544
@NickYoungPeng

**Review by TyrantLucifer — COMMENTED** [review]
Overall LGTM, PTAL @ic4y @liugddx

**Comment by NickCodeJourney** [comment]
@zhilinli123 TBR

File: docs/en/connector-v2/sink/Kingbase.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+## Key Features
+
+- [x] [exactly-once](../../concept/connector-v2-features.md)
+- [ ] [cdc](../../concept/connector-v2-features.md)
+
+## Description
+
+> Use `Xa transactions` to ensure `exactly-once`. So only support `exactly-once` for the database which is
+> support `Xa transactions`. You can set `is_exactly_once=true` to enable it.
```
Comment (liugddx) [inline docs/en/connector-v2/sink/Kingbase.md]: @ic4y Does kingbase have this feature?

File: docs/en/connector-v2/sink/Kingbase.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+## Supported DataSource Info
+
+| Datasource | Supported versions |        Driver        |                   Url                    |                                             Maven                                              |
+|------------|--------------------|----------------------|------------------------------------------|------------------------------------------------------------------------------------------------|
+| Kingbase   | v8r6               | com.kingbase8.Driver | jdbc:kingbase8://localhost:54321/db_test | [Download](https://repo1.maven.org/maven2/cn/com/kingbase/kingbase8/8.6.0/kingbase8-8.6.0.jar) |
+
+## Database Dependency
```
Comment (liugddx) [inline docs/en/connector-v2/sink/Kingbase.md:31]: Add kingbase support versions? @zhilinli123

**Comment by liugddx** [comment]
You need to put the screenshot of the e2e test here

File: docs/en/connector-v2/sink/Kingbase.md (hunk 3 above)
Comment (zhilinli123) [reply docs/en/connector-v2/sink/Kingbase.md:31]: > Add kingbase support versions? @zhilinli123

yes cc @liugddx @NickYoungPeng  https://github.com/apache/seatunnel/pull/5249

File: docs/en/connector-v2/sink/Kingbase.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# Kingbase
+
+> JDBC Kingbase Sink Connector
+
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/Kingbase.md:4]: reference: https://github.com/apache/seatunnel/pull/5249

File: docs/en/connector-v2/sink/Kingbase.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+  FakeSource {
+    parallelism = 1
+    result_table_name = "fake"
+    row.num = 16
+    schema = {
+      fields {
+        name = "string"
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/Kingbase.md]: Add a full type test

File: docs/en/connector-v2/source/Kingbase.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
+# Kingbase
+
+> JDBC Kingbase Source Connector
+
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/Kingbase.md:4]: Add version information as above

File: docs/en/connector-v2/source/Kingbase.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+> It is more efficient to read your data source according to the upper and lower boundaries you configured
+
+```
+source {
+  Jdbc {
+    driver = "com.kingbase8.Driver"
+    url = "jdbc:kingbase8://localhost:54321/db_test"
+    user = "root"
+    password = ""
+    compatible_mode = "mysql"
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/Kingbase.md]: mykingbase?

File: docs/en/connector-v2/sink/Kingbase.md (hunk 2 above)
Comment (NickCodeJourney) [reply docs/en/connector-v2/sink/Kingbase.md]: i will test the feature

Comment (NickCodeJourney) [reply docs/en/connector-v2/sink/Kingbase.md]: > @ic4y Does kingbase have this feature?

sorry kingbase don't support the feature

Comment (NickCodeJourney) [reply docs/en/connector-v2/sink/Kingbase.md]: `com.kingbase8.util.KSQLException: ERROR: prepared transactions are disabled
  Hint: Set max_prepared_transactions to a nonzero value.`
need Set max_prepared_transactions to a nonzero value

Comment (NickCodeJourney) [reply docs/en/connector-v2/sink/Kingbase.md]: image
i try it but not ok

Comment (NickCodeJourney) [reply docs/en/connector-v2/sink/Kingbase.md]: don't support

File: docs/en/connector-v2/sink/Kingbase.md (hunk 5 above)
Comment (zhilinli123) [reply docs/en/connector-v2/sink/Kingbase.md]: such as: @NickYoungPeng 
```
source {
  FakeSource {
    row.num = 10
    map.size = 10
    array.size = 10
    bytes.length = 10
    string.length = 10
    schema = {
      fields {
        c_map = "map<string, array<int>>"
        c_array = "array<int>"
        c_string = string
        c_boolean = boolean
        c_tinyint = tinyint
        c_smallint = smallint
        c_int = int
        c_bigint = bigint
        c_float = float
        c_double = double
        c_decimal = "decimal(16, 1)"
        c_null = "null"
        c_bytes = bytes
        c_date = date
        c_timestamp = timestamp
      }
    }
    }
}
```

**Comment by zhilinli123** [comment]
Wait for ci

**Comment by NickCodeJourney** [comment]
> You need to put the screenshot of the e2e test here

it's above and it's folded

**Comment by NickYoungPeng** [comment]
> Wait for ci

last ci `Error:  Client bundle compiled with errors therefore further build is impossible.`

**Comment by NickYoungPeng** [comment]
> Wait for ci

`SyntaxError: /home/runner/work/seatunnel/seatunnel/seatunnel-website/docs/connector-v2/sink/Kingbase.md: Expected corresponding JSX closing tag for <td>. (76:55)
Error:  Client bundle compiled with errors therefore further build is impossible.
  74 | </tr>
  75 | <tr parentName="tbody">
> 76 | <td parentName="tr" {...{"align":null}}>{`SMALLSERIAL`}</br>{` SERIAL `}</br>{`INT4`}</td>
     |                                                        ^
  77 | <td parentName="tr" {...{"align":null}}>{`INT`}</td>
  78 | </tr>
  79 | <tr parentName="tbody">
SyntaxError: /home/runner/work/seatunnel/seatunnel/seatunnel-website/docs/connector-v2/source/Kingbase.md: Expected corresponding JSX closing tag for <td>. (77:55)
  75 | </tr>
  76 | <tr parentName="tbody">
> 77 | <td parentName="tr" {...{"align":null}}>{`SMALLSERIAL`}</br>{` SERIAL `}</br>{`INT4`}</td>
     |                                                        ^
  78 | <td parentName="tr" {...{"align":null}}>{`INT`}</td>
  79 | </tr>
  80 | <tr parentName="tbody">
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! seatunnel-website@0.0.0 build: `docusaurus build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the seatunnel-website@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-08-10T02_06_00_437Z-debug.log
Error: Process completed with exit code 1.`

**Comment by NickCodeJourney** [comment]
> You need to put the screenshot of the e2e test here

https://github.com/apache/seatunnel/pull/4803#issuecomment-1634345386
https://github.com/apache/seatunnel/pull/4803#issuecomment-1632798856

**Review by liugddx — APPROVED** [review]
LGTM

**Comment by liugddx** [comment]
> LGTM

Please screenshot the test results. PTAL @ic4y

**Review by zhilinli123 — APPROVED** [review]

**Comment by NickCodeJourney** [comment]
> 



> > LGTM
> 
> Please screenshot the test results. PTAL @ic4y

Snipaste_2023-08-15_11-19-09
Snipaste_2023-08-15_11-18-02

**Comment by NickYoungPeng** [comment]
close #4758 #4512

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**